compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -877,41 +876,70 @@ impl<'context> Elaborator<'context> {
         self.file = trait_impl.file_id;
         self.local_module = trait_impl.module_id;
 
-        let unresolved_type = trait_impl.object_type;
-        let self_type_span = unresolved_type.span;
         let old_generics_length = self.generics.len();
         self.generics = trait_impl.resolved_generics;
+        self.current_trait_impl = trait_impl.impl_id;
 
-        let trait_generics =
-            vecmap(&trait_impl.trait_generics, |generic| self.resolve_type(generic.clone()));
+        self.elaborate_functions(trait_impl.methods);
 
-        let self_type = trait_impl.resolved_object_type.unwrap_or(Type::Error);
+        self.self_type = None;
+        self.current_trait_impl = None;
+        self.generics.truncate(old_generics_length);
```

Comment:
`recover_generics`

Reply:
Realizing this may be outdated now

Reply:
I've changed this one to a `.clear()` since we're assigning to `self.generics` instead of just pushing to it.